solidity/contracts/converter/types/standard-pool/StandardPoolConverter.sol

```diff
@@ -0,0 +1,1386 @@
+     * @param _conversionFee   conversion fee in units of the target token
+     */
+    event Conversion(
+        IERC20Token indexed _fromToken,
+        IERC20Token indexed _toToken,
+        address indexed _trader,
+        uint256 _amount,
+        uint256 _return,
+        int256 _conversionFee
+    );
+
+    /**
+     * @dev triggered when the rate between two tokens in the converter changes
+     * note that the event might be dispatched for rate updates between any two tokens in the converter
+     *
+     * @param  _token1 address of the first token
+     * @param  _token2 address of the second token
+     * @param  _rateN  rate of 1 unit of `_token1` in `_token2` (numerator)
+     * @param  _rateD  rate of 1 unit of `_token1` in `_token2` (denominator)
+     */
+    event TokenRateUpdate(IERC20Token indexed _token1, IERC20Token indexed _token2, uint256 _rateN, uint256 _rateD);
+
+    /**
+     * @dev triggered when the conversion fee is updated
+     *
+     * @param  _prevFee    previous fee percentage, represented in ppm
+     * @param  _newFee     new fee percentage, represented in ppm
+     */
+    event ConversionFeeUpdate(uint32 _prevFee, uint32 _newFee);
+
+    /**
+     * @dev triggered after liquidity is added
+     *
+     * @param  _provider       liquidity provider
+     * @param  _reserveToken   reserve token address
+     * @param  _amount         reserve token amount
+     * @param  _newBalance     reserve token new balance
+     * @param  _newSupply      pool token new supply
+     */
+    event LiquidityAdded(
+        address indexed _provider,
+        IERC20Token indexed _reserveToken,
+        uint256 _amount,
+        uint256 _newBalance,
+        uint256 _newSupply
+    );
+
+    /**
+     * @dev triggered after liquidity is removed
+     *
+     * @param  _provider       liquidity provider
+     * @param  _reserveToken   reserve token address
+     * @param  _amount         reserve token amount
+     * @param  _newBalance     reserve token new balance
+     * @param  _newSupply      pool token new supply
+     */
+    event LiquidityRemoved(
+        address indexed _provider,
+        IERC20Token indexed _reserveToken,
+        uint256 _amount,
+        uint256 _newBalance,
+        uint256 _newSupply
+    );
+
+    /**
+     * @dev initializes a new StandardPoolConverter instance
+     *
+     * @param  _anchor             anchor governed by the converter
+     * @param  _registry           address of a contract registry contract
+     * @param  _maxConversionFee   maximum conversion fee, represented in ppm
+     */
+    constructor(
+        IConverterAnchor _anchor,
+        IContractRegistry _registry,
+        uint32 _maxConversionFee
+    ) public ContractRegistryClient(_registry) validAddress(address(_anchor)) validConversionFee(_maxConversionFee) {
+        anchor = _anchor;
+        maxConversionFee = _maxConversionFee;
+    }
+
+    // ensures that the converter is active
+    modifier active() {
```

Comment:
You can also consider creating another base class for this, thus preventing the duplications (but it's not a priority, of course)

Reply:
Will be done separately, as part of a "clean" task.